src/editor/Gui.cpp

```diff
@@ -2328,13 +2449,34 @@ void Gui::drawSettings() {
 
 			if (ImGui::Button("Apply Changes")) {
 				g_settings.gamedir = string(gamedir);
+				g_settings.workingdir = string(workingdir);
 				/* fixup gamedir */
 				if (g_settings.gamedir.size() && (g_settings.gamedir[g_settings.gamedir.size() - 1] == '/'
 					|| g_settings.gamedir[g_settings.gamedir.size() - 1] == '\\'))
 				{
 					g_settings.gamedir.pop_back();
-					sprintf(gamedir, "%s", g_settings.gamedir.c_str());
 				}
+				sprintf(gamedir, "%s", g_settings.gamedir.c_str());
+
+				/* fixup workingdir */
+				if (g_settings.workingdir.size() > 1)
+				{
+					if (g_settings.workingdir[0] != '\\' &&
+						g_settings.workingdir[0] != '/')
+					{
+						g_settings.workingdir = "/" + g_settings.workingdir;
+					}
+					if (g_settings.workingdir[g_settings.workingdir.size() - 1] != '\\' &&
+						g_settings.workingdir[g_settings.workingdir.size() - 1] != '/')
+					{
+						g_settings.workingdir = g_settings.workingdir + "/";
+					}
+				}
+				else
+				{
+					g_settings.workingdir = "/";
+				}
+				sprintf(workingdir, "%s", g_settings.workingdir.c_str());

```

Comment:
Another reason to create a method that can joins paths, so that it doesn't matter if there are slashes or not in the setting.

Reply:
Yes))